File: 002/soln.ts
/*
Solution to Project Euler problems by Joseph Jude

Webpage: http://tech.jjude.com/euler
Git Repo: https://github.com/jjude/euler 

execute with: tsc soln.ts && node soln.js

Problem 002: Sum of even valued Fibonacci numbers less than 4 million
WIL: tuple assignment
*/

var answer = 0;
var current_fibonacci = 1
var next_fibonacci = 2

while (current_fibonacci <= 4000000) {
  if (current_fibonacci % 2 == 0) {
    answer += current_fibonacci;
  }
  [current_fibonacci, next_fibonacci] = [next_fibonacci, current_fibonacci + next_fibonacci]   
}
console.log(answer)
// 4613732

File: 002/soln.swift
/*
Solution to Project Euler problems by Joseph Jude

Webpage: http://tech.jjude.com/euler
Git Repo: https://github.com/jjude/euler

execute with: swift soln.swift
compile with: swiftc soln.swift

Problem 002: Sum of even valued Fibonacci numbers less than 4 million
WIL: tuple assignment
*/

var answer = 0
var currentFibonacci = 1
var nextFibonacci = 2

while currentFibonacci <= 4000000 {
  if currentFibonacci % 2 == 0 {
    answer += currentFibonacci
  }
  (currentFibonacci, nextFibonacci) = (nextFibonacci, currentFibonacci + nextFibonacci)
}
print(answer)
// 4613732
